### PR TITLE
refactor arithmetic check

### PR DESCRIPTION
**Pull request by charlesjhongc**

### CODE REVIEW

File: contracts/AllowanceTarget.sol [hunk 1]
```diff
@@ -16,8 +16,11 @@ contract AllowanceTarget is IAllowanceTarget, Pausable, Ownable {
 
     constructor(address _owner, address[] memory trustedCaller) Ownable(_owner) {
         uint256 callerCount = trustedCaller.length;
-        for (uint256 i = 0; i < callerCount; ++i) {
+        for (uint256 i = 0; i < callerCount; ) {
             authorized[trustedCaller[i]] = true;
+            unchecked {
+                ++i;
+            }
```
Comment (pilagod) [inline contracts/AllowanceTarget.sol:23]: I feel that this is a bit too much for gas optimization. I would prefer trading this gas cost for coding convention and readability.

File: contracts/LimitOrderSwap.sol [hunk 2]
```diff
@@ -82,7 +82,7 @@ contract LimitOrderSwap is ILimitOrderSwap, Ownable, TokenCollector, EIP712, Ree
         uint256[] memory takerTokenAmounts = new uint256[](orders.length);
         uint256 wethToPay;
         address payable _feeCollector = feeCollector;
-        for (uint256 i = 0; i < orders.length; ++i) {
+        for (uint256 i = 0; i < orders.length; ) {
```
Comment (pilagod) [inline contracts/LimitOrderSwap.sol:85]: nit: Shall we cache the `orders.length` and also other loops in this PR?

File: contracts/AllowanceTarget.sol (hunk 1 above)
Comment (NIC619) [reply contracts/AllowanceTarget.sol:23]: I feel like this pattern is more difficult to follow too. Unless documented properly, devs could misuse or break the pattern when copy-pasting it in the future.

Comment (charlesjhongc) [reply contracts/AllowanceTarget.sol:23]: Ok, let's skip this refactoring for now. It's kind of common for me but I do think it's more error-prone.

File: contracts/LimitOrderSwap.sol (hunk 2 above)
Comment (charlesjhongc) [reply contracts/LimitOrderSwap.sol]: Will do in another PR.

File: contracts/AllowanceTarget.sol (hunk 1 above)
Comment (charlesjhongc) [reply contracts/AllowanceTarget.sol]: Fixed.

**Review by NIC619 — APPROVED** [review]

**Review by pilagod — APPROVED** [review]

**Comment by charlesjhongc** [comment]
Fix array length cache in #268

**Pull request merged**